app/server.R

```diff
@@ -51,7 +51,7 @@ server <- function(input, output, session) {
     # hit the API
     # do not proceed if go was not clicked at all
     validate(need(input$go > 0, ""))
-    validate(need(nchar(input$zip) >= 5, ""))
+    validate(need((input$country == "us" & nchar(input$zip) >= 5) | (input$country == "be" & nchar(input$zip) >= 4), ""))
```

Comment:
Why are we asking nchar(input$zip) to be >=5, why not nchar(input$zip) == 5? Same for the BE zip code.

Reply:
Good point. I think the original idea was to leverage the API error message saying that the zip code parameter must be 5 digits if someone inputs more than 5 digits for a us zip code. 

We added `nchar(input$zip) >= 5` to avoid the RShiny app from calling the API before the person might be done typing in their zip code.  If we changed the program to `nchar(input$zip) == 5` then the RShiny app will not call the API if >5 digits are entered and no error message will be returned to the user telling them that the zip code parameter must be 5 digits.